src/qiskit_iqm/iqm_provider.py

```diff
@@ -138,6 +139,50 @@ def serialize_circuit(self, circuit: QuantumCircuit) -> Circuit:
         return Circuit(name=circuit.name, instructions=instructions, metadata=metadata)
 
 
+class IQMFacadeBackend(IQMBackend):
+    """Facade backend for mimicking the execution of quantum circuits on IQM quantum computers. Allows to submit a
+     circuit to the IQM server, and if the execution was successful, performs a simulation with a respective IQM noise
+     model locally, then returns the simulated results.
+
+    Args:
+        client: client instance for communicating with an IQM server
+        **kwargs: optional arguments to be passed to the parent Backend initializer
+    """
+
+    def __init__(self, client: IQMClient, **kwargs):
+        self.fake_adonis = IQMFakeAdonis()
+        target_architecture = client.get_quantum_architecture()
+
+        if not self.fake_adonis.validate_compatible_architecture(target_architecture):
+            raise ValueError('Quantum architecture of the remote quantum computer does not match Adonis.')
+
+        super().__init__(client, **kwargs)
+        self.client = client
+
+    def _validate_no_empty_cregs(self, circuit):
+        """Returns True if given circuit has no empty (unused) classical registers, False otherwise."""
+        cregs_utilization = dict.fromkeys(circuit.cregs, 0)
+        used_cregs = [circuit.find_bit(i.clbits[0]).registers[0][0] for i in circuit.data if len(i.clbits) > 0]
+        for creg in used_cregs:
+            cregs_utilization[creg] += 1
+
+        if 0 in cregs_utilization.values():
+            return False
+        return True
+
+    def run(self, run_input: Union[QuantumCircuit, list[QuantumCircuit]], **options) -> JobV1:
+        circuits = [run_input] if isinstance(run_input, QuantumCircuit) else run_input
+        circuits_validated_cregs: list[bool] = [self._validate_no_empty_cregs(circuit) for circuit in circuits]
+        if not all(circuits_validated_cregs):
+            raise ValueError('One or more circuits contain unused classical registers.')
```

Comment:
The error could be a bit more informative since it is not really intuitive why this is necessary in my opinion.

Reply:
I'm having trouble formulating this in a short, error-style message form. So far, I came up with `'One or more circuits contain unused classical registers, which leads to 0-filled registers in results'` ([commit](https://github.com/iqm-finland/qiskit-on-iqm/pull/53/commits/673a8addf8f082cf9eb0e813887059758546204d)). Do you have any suggestions?

Reply:
Maybe `One or more circuits contain unused classical registers. This is not allowed for Facade simulation, see user guide.` And then we add something to the user guide?

Reply:
Good idea, [done](https://github.com/iqm-finland/qiskit-on-iqm/pull/53/commits/f9c68de73c4cf06e58a8d20fd5a692dcdc043fb9).